streams/prngs/testu01-prngs/testu01_factory.cc

```diff
@@ -0,0 +1,28 @@
+//
+// Created by mhajas on 11/14/17.
+//
+#include <memory>
+#include <streams/prngs/prng_interface.h>
+#include <eacirc-core/json.h>
+#include <memory.h>
+//#include <streams/prngs/testu01-prngs/functions/unif01/dummy_generator.h>
+#include <streams/prngs/testu01-prngs/functions/ulcg/ulcg_generator.h>
+#include <streams/prngs/testu01-prngs/functions/umrg/umrg_generator.h>
+#include <streams/prngs/testu01-prngs/functions/uxorshift/uxorshift_generator.h>
+
+namespace prng {
+
+
+    std::unique_ptr<prng_interface>
+    create_testu01_interface(const json& configuration, default_seed_source& seeder, std::unordered_map<std::string, std::shared_ptr<std::unique_ptr<stream>>> &pipes) {
+        std::string name = configuration.at("algorithm");
+
+        if(name == "testu01-ulcg") return std::make_unique<ulcg_generator>(configuration, seeder, pipes);
+        if(name == "testu01-umrg") return std::make_unique<umrg_generator>(configuration, seeder, pipes);
+        if(name == "testu01-uxorshift") return std::make_unique<uxorshift_generator>(configuration, seeder, pipes);
```

Comment:
Document on wiki and write, what has to be done to continue with other functions